File: src/modules/telegram.py

import requests

def _call_method (method, bot_token, data = None, files = None):
	return requests.post(
		'https://api.telegram.org/bot{}/{}'.format(bot_token, method),
		data = data,
		files = files,
	)

def send (credentials = None, text = None, log_file_name = None):
	chat_id = credentials['chat_id']
	bot_token = credentials['bot_token']

	response = _call_method(
		'sendMessage',
		bot_token = bot_token,
		data = {
			'chat_id': chat_id,
			'text': text,
			'parse_mode': 'markdown',
		},
	)
	# print('sendMessage status code:', response.status_code)
	# print('sendMessage body:', response.json())

	message_id = response.json()['result']['message_id']

	if log_file_name:
		files = {
			'document': (
				'log.txt',
				open(log_file_name, 'rb'),
			),
		}

		response = _call_method(
			'sendDocument',
			bot_token = bot_token,
			data = {
				'chat_id': chat_id,
				'reply_to_message_id': message_id,
			},
			files = files,
		)
		# print('sendDocument status code:', response.status_code)
		# print('sendDocument body:', response.json())

File: src/run/notification/telegram.js

import { file_name as log_file_name } from '../../run/logger.file.js';
import { httpPostRequest }                from '../../utils.js';

async function callAPI(
	method,
	bot_token,
	data = {},
	files = {},
) {
	return httpPostRequest({
		url: `https://api.telegram.org/bot${bot_token}/${method}`,
		data,
		files,
	});
}

export default async function (
	{
		bot_token,
		chat_id,
	},
	text,
	add_log = false,
) {
	const response_message = await callAPI(
		'sendMessage',
		bot_token,
		{
			chat_id,
			text,
			parse_mode: 'markdown',
		},
	);
	if (process.env.SUPERNOVA_DEBUG) {
		console.log('response_message', response_message);
	}

	if (add_log) {
		const { message_id } = response_message.body.result;

		const response_file = await callAPI(
			'sendDocument',
			bot_token,
			{
				chat_id,
				reply_to_message_id: message_id,
			},
			{
				document: {
					path: log_file_name,
					name: 'log.txt',
				},
			},
		);

		if (process.env.SUPERNOVA_DEBUG) {
			console.log('response_file', response_file);
		}
	}
}
